src/pages/namespace/Explorer/Tree/components/modals/CreateNew/Gateway/Route.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  DialogClose,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "~/design/Dialog";
+import { Network, PlusCircle } from "lucide-react";
+import { SubmitHandler, useForm } from "react-hook-form";
+
+import Button from "~/design/Button";
+import FormErrors from "~/componentsNext/FormErrors";
+import Input from "~/design/Input";
+import { addYamlFileExtension } from "../../../../utils";
+import { defaultEndpointFileYaml } from "~/pages/namespace/Explorer/Endpoint/EndpointEditor/utils";
+import { fileNameSchema } from "~/api/tree/schema/node";
+import { pages } from "~/util/router/pages";
+import { useCreateWorkflow } from "~/api/tree/mutate/createWorkflow";
+import { useNamespace } from "~/util/store/namespace";
+import { useNavigate } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+type FormInput = {
+  name: string;
+  fileContent: string;
+};
+
+const NewRoute = ({
+  path,
+  close,
+  unallowedNames,
+}: {
+  path?: string;
+  close: () => void;
+  unallowedNames?: string[];
+}) => {
+  const { t } = useTranslation();
+  const namespace = useNamespace();
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { isDirty, errors, isValid, isSubmitted },
+  } = useForm<FormInput>({
+    resolver: zodResolver(
```

Comment:
This would be a bit nicer to read if we declared a `const resolver` separately, instead of nesting it here.

Reply:
I agree! The problem is that it has a dependency to the `t` function to `unallowedNames`. So I think I will make it a separate function/hook then? I think we had this discussion before and decided against it. If you search for `unallowedNames` you will find a lot of similar resolver.

Reply:
What I mean isn't a big change, the const would still be declared in the same scope and have access to all variables. It would just avoid one layer of nesting / indentation.

Not sure but I think I remember a case where the resolver depends on values returned by the useForm() hook but that doesn't seem to be the case here? 

```
  const resolver = zodResolver(
    z.object({
      name: fileNameSchema
        .transform((enteredName) => addYamlFileExtension(enteredName))
        .refine(
          (nameWithExtension) =>
            !(unallowedNames ?? []).some(
              (unallowedName) => unallowedName === nameWithExtension
            ),
          {
            message: t("pages.explorer.tree.newRoute.nameAlreadyExists"),
          }
        ),
      fileContent: z.string(),
    })
  );

  const {
    register,
    handleSubmit,
    formState: { isDirty, errors, isValid, isSubmitted },
  } = useForm<FormInput>({
    resolver,
    defaultValues: {
      fileContent: defaultEndpointFileYaml,
    },
  });
  ```

Reply:
Gotcha. I will resolve it here and make another PR